store/providers/xorm/certificate_types.go

```diff
@@ -26,22 +29,51 @@ func (Certificate) TableName() string {
 	return "ac_cluster_certificate"
 }
 
-func encodeCertificate(crt *x509.Certificate, key *rsa.PrivateKey) *Certificate {
+func (certificate *Certificate) FillCertFields(name, uid, clusterName string, clusterId, createdAt int64) {
+	certificate.Name = name
+	certificate.UID = uid
+	certificate.ClusterName = clusterName
+	certificate.ClusterID = clusterId
+	certificate.CreatedUnix = createdAt
+}
+
+func EncodeCertificate(crt *x509.Certificate, key *rsa.PrivateKey) (*Certificate, error) {
+	secretId := types.RotateQuarterly()
+	certCipher, err := encryptData(secretId, cert.EncodeCertPEM(crt))
+	if err != nil {
+		return nil, fmt.Errorf("failed to encrypt: %v", err)
```

Comment:
log the error, then return
```suggestion
                log.Error("failed to encrypt certificate", err)
		return nil, err
```